webapp/packages/plugin-data-spreadsheet-new/src/DataGrid/DataGridTable.tsx

```diff
@@ -164,18 +166,49 @@ export const DataGridTable = observer<IDataPresentationProps>(function DataGridT
     }
   }
 
-  const hamdlers = useObjectRef(() => ({
+  const handlers = useObjectRef(() => ({
     selectCell(pos: Position, scroll = false): void {
       if (dataGridRef.current?.selectedCell.idx !== pos.idx || dataGridRef.current.selectedCell.rowIdx !== pos.rowIdx || scroll) {
         dataGridRef.current?.selectCell(pos);
       }
     },
+    focusCell(key: Partial<IResultSetElementKey> | null, initial = false) {
+      // TODO: we need this delay to update focus after render rows update
+      setTimeout(() => {
+        if ((!key?.column || !key?.row) && initial) {
+          const selectedElements = selectionAction.getSelectedElements();
+
+          if (selectedElements.length > 0) {
+            key = selectedElements[0];
+          } else {
+            key = { column: viewAction.columnKeys[0], row: viewAction.rowKeys[0] };
+          }
+        }
+
+        if (!key?.column || !key?.row) {
+          if (initial) {
+            focusSyncRef.current = { idx: 0, rowIdx: -1 };
+            this.selectCell(focusSyncRef.current);
+          } else {
+            focusSyncRef.current = null;
+          }
+          return;
+        }
+
+        const idx = tableData.getColumnIndexFromColumnKey(key.column!);
+        const rowIdx = tableData.getRowIndexFromKey(key.row!);
+
+        focusSyncRef.current = { idx, rowIdx };
+
+        this.selectCell({ idx, rowIdx });
```

Comment:
```suggestion
        if (!key?.column || !key?.row) {
          if (initial) {
            const selectedElements = selectionAction.getSelectedElements();

            if (selectedElements.length > 0) {
              key = selectedElements[0];
            } else {
              key = { column: viewAction.columnKeys[0], row: viewAction.rowKeys[0] };
            }

            focusSyncRef.current = { idx: 0, rowIdx: -1 };
            this.selectCell(focusSyncRef.current);
          } else {
            focusSyncRef.current = null;
          }
        }

        if (!key?.column || !key?.row) {
          return; // TODO throw an error that cannot focus for some reason?
        }

        const idx = tableData.getColumnIndexFromColumnKey(key.column);
        const rowIdx = tableData.getRowIndexFromKey(key.row);

        focusSyncRef.current = { idx, rowIdx };

        this.selectCell({ idx, rowIdx });
```

Reply:
I suggest to simplify a bit if/else logic + let's forbid to use `!` operator in most cases. because this way we trick TS and make potential points for the bugs to appear. It is better to handle all edge cases then use this operator

Reply:
actually in current implementation `!` can be omitted also

Reply:
and your implementation is incorrect